libcudacxx/.upstream-tests/test/std/iterators/iterator.requirements/indirectcallable/indirectinvocable/indirect_binary_predicate.compile.pass.cpp

```diff
@@ -33,6 +33,10 @@ struct GoodPredicate {
 // Should work when all constraints are satisfied
 static_assert(cuda::std::indirect_binary_predicate<GoodPredicate<It1, It2>, It1, It2>);
 static_assert(cuda::std::indirect_binary_predicate<bool(*)(int, float), int*, float*>);
+
+#ifdef TEST_COMPILER_CLANG_CUDA
+#pragma clang diagnostic ignored "-Wunneeded-internal-declaration"
+#endif // TEST_COMPILER_CLANG_CUDA
```

Comment:
Seems like we should have a `_LIBCUDACXX_DISABLE_CLANG_CUDA_DIAGNOSTIC` macro?

Reply:
I believe we need one for gcc / clang diagnostics. I can cook something up, but it should not be required for this PR